test-e2e/android/app/src/androidTest/java/com/mopro/mopro_app/ExampleInstrumentedTest.kt

```diff
@@ -61,11 +66,16 @@ class ExampleInstrumentedTest {
         composeTestRule.onNodeWithTag("halo2GenerateProofButton").performClick()
         composeTestRule.onNodeWithTag("halo2GenerateProofButton").assertIsDisplayed()
 
-        // Test click circom verify proof button
-        // Wait until the second button is enabled
-        Thread.sleep(500)
+        // Wait up to 5 seconds for any halo2VerifyProofButton to become enabled before continuing the test
+        composeTestRule.waitUntil(timeoutMillis = 5000) {
+            try {
+                composeTestRule.onNodeWithTag("halo2VerifyProofButton").assertIsEnabled()
+                true
+            } catch (e: Exception) {
+                false
+            }
+        }
 
-        composeTestRule.onNodeWithTag("halo2VerifyProofButton").performClick()
```

Comment:
this should be kept
so that it throws error when `halo2VerifyProofButton` fails